### PR TITLE
feat!: return unified JSON responses on various requests

### PR DESCRIPTION
**Pull request by asafMasa**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✔                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

File: openapi3.yaml [hunk 1]
```diff
@@ -29,8 +29,6 @@ paths:
                 type: array
                 items:
                   $ref: '#/components/schemas/jobResponse'
-        204:
```
Comment (cl-shaharshavit-rnd) [inline openapi3.yaml:32]: if that was actually implemented removing it is breaking change

File: openapi3.yaml [hunk 2]
```diff
@@ -207,8 +208,6 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/tasksResponse'
-        204:
```
Comment (cl-shaharshavit-rnd) [inline openapi3.yaml:210]: if that was actually implemented removing it is breaking change

File: openapi3.yaml [hunk 3]
```diff
@@ -340,8 +341,6 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/taskStatus'
-        204:
```
Comment (cl-shaharshavit-rnd) [inline openapi3.yaml:343]: if that was actually implemented removing it is breaking change

File: openapi3.yaml [hunk 4]
```diff
@@ -471,8 +471,14 @@ paths:
       operationId: abortTasksAndJob
       summary: updates job and pending tasks statuses to "Aborted"
       responses:
-        204:
```
Comment (cl-shaharshavit-rnd) [inline openapi3.yaml:474]: if that was actually implemented removing it is breaking change

File: tests/integration/jobs/jobs.spec.ts [hunk 5]
```diff
@@ -374,7 +374,7 @@ describe('jobs', function () {
         const response = await requestSender.getResources(filter);
         expect(response).toSatisfyApiSpec();
 
-        expect(response.status).toBe(httpStatusCodes.NO_CONTENT);
+        expect(response.status).toBe(httpStatusCodes.OK);
```
Comment (cl-shaharshavit-rnd) [inline tests/integration/jobs/jobs.spec.ts:377]: you added specific message for every action test that the correct value is returned

File: tests/integration/jobs/tasks.spec.ts [hunk 6]
```diff
@@ -170,14 +170,14 @@ describe('tasks', function () {
...
-    it('should return 204 for job without tasks', async function () {
+    it('should return 200 for job without tasks', async function () {
       const jobsFindMock = taskRepositoryMocks.findMock;
       jobsFindMock.mockResolvedValue([] as TaskEntity[]);
 
       const response = await requestSender.getAllResources(jobId);
       expect(response).toSatisfyApiSpec();
 
-      expect(response.status).toBe(httpStatusCodes.NO_CONTENT);
+      expect(response.status).toBe(httpStatusCodes.OK);
```
Comment (cl-shaharshavit-rnd) [inline tests/integration/jobs/tasks.spec.ts:180]: you added specific message for every action test that the correct value is returned

File: tests/integration/taskManagement/taskManagement.spec.ts [hunk 7]
```diff
@@ -254,7 +254,7 @@ describe('tasks', function () {
 
         const response = await requestSender.abortJobAndTasks(jobId);
 
-        expect(response.status).toBe(httpStatusCodes.NO_CONTENT);
+        expect(response.status).toBe(httpStatusCodes.OK);
         expect(jobRepositoryMocks.saveMock).toHaveBeenCalledTimes(1);
```
Comment (cl-shaharshavit-rnd) [inline tests/integration/taskManagement/taskManagement.spec.ts:258]: you added specific message for every action test that the correct value is returned

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]
pr name should be changed to:
"feat!: return unified JSON responses on various requests"
as it has breaking changes

File: tests/integration/jobs/jobs.spec.ts (hunk 5 above)
Comment (asafMasa) [reply tests/integration/jobs/jobs.spec.ts:379]: fixed

File: tests/integration/jobs/tasks.spec.ts (hunk 6 above)
Comment (asafMasa) [reply tests/integration/jobs/tasks.spec.ts:182]: fixed

File: tests/integration/taskManagement/taskManagement.spec.ts (hunk 7 above)
Comment (asafMasa) [reply tests/integration/taskManagement/taskManagement.spec.ts:260]: fixed

File: openapi3.yaml (hunk 1 above)
Comment (cl-shaharshavit-rnd) [reply openapi3.yaml:32]: I didnt say it wrong.
I said its breaking and you should create tasks for fixing the broken client

File: openapi3.yaml (hunk 2 above)
Comment (cl-shaharshavit-rnd) [reply openapi3.yaml:210]: I didnt say it wrong.
I said its breaking and you should create tasks for fixing the broken client

**Review by cl-shaharshavit-rnd — APPROVED** [review]

**Review by shimoncohen — APPROVED** [review]

**Review by RonitKissis — APPROVED** [review]

**Pull request merged**